core/src/main/java/com/adyen/checkout/core/model/Amount.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2025 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 30/4/2025.
+ */
+
+package com.adyen.checkout.core.model
+
+import com.adyen.checkout.core.exception.ModelSerializationException
+import com.adyen.checkout.core.internal.data.model.ModelObject
+import com.adyen.checkout.core.internal.data.model.getLongOrNull
+import com.adyen.checkout.core.internal.data.model.getStringOrNull
+import kotlinx.parcelize.Parcelize
+import org.json.JSONException
+import org.json.JSONObject
+
+@Parcelize
+data class Amount(
+    var currency: String? = null,
```

Comment:
It would be a good time to convert all these mutable objects to immutable objects with `val`. WDYT?